src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionSerializer.cs

```diff
@@ -0,0 +1,192 @@
+            value = this._settingManager.GetValueOrDefault(storageKey, optionKey.Option.DefaultValue);
+
+            // VS's ISettingsManager has some quirks around storing enums.  Specifically,
+            // it *can* persist and retrieve enums, but only if you properly call 
+            // GetValueOrDefault<EnumType>.  This is because it actually stores enums just
+            // as ints and depends on the type parameter passed in to convert the integral
+            // value back to an enum value.  Unfortunately, we call GetValueOrDefault<object>
+            // and so we get the value back as boxed integer.
+            //
+            // Because of that, manually convert the integer to an enum here so we don't
+            // crash later trying to cast a boxed integer to an enum value.
+            if (optionKey.Option.Type.IsEnum)
+            {
+                if (value != null)
+                {
+                    value = Enum.ToObject(optionKey.Option.Type, value);
+                }
+            }
+            else if (optionKey.Option.Type == typeof(CodeStyleOption<bool>))
+            {
+                // We store these as strings, so deserialize
+                var serializedValue = value as string;
+
+                if (serializedValue != null)
+                {
+                    value = CodeStyleOption<bool>.FromXElement(XElement.Parse(serializedValue));
+                }
+                else
+                {
+                    value = optionKey.Option.DefaultValue;
+                }
+            }
+
+            return true;
+        }
+
+        private void RecordObservedValueToWatchForChanges(OptionKey optionKey, string storageKey)
+        {
+            // We're about to fetch the value, so make sure that if it changes we'll know about it
+            lock (_optionsToMonitorForChangesGate)
+            {
+                var optionKeysToMonitor = _optionsToMonitorForChanges.GetOrAdd(storageKey, _ => new List<OptionKey>());
+
+                if (!optionKeysToMonitor.Contains(optionKey))
+                {
+                    optionKeysToMonitor.Add(optionKey);
+                }
+            }
+        }
+
+        public bool TryPersist(OptionKey optionKey, object value)
+        {
+            if (this._settingManager == null)
+            {
+                Debug.Fail("Manager field is unexpectedly null.");
+                return false;
+            }
+
+            // Do we roam this at all?
+            var roamingSerialization = optionKey.Option.Serializations.OfType<RoamingProfileSerialization>().SingleOrDefault();
+
+            if (roamingSerialization == null)
+            {
+                value = null;
+                return false;
+            }
+
+            var storageKey = roamingSerialization.GetKeyNameForLanguage(optionKey.Language);
+
+            RecordObservedValueToWatchForChanges(optionKey, storageKey);
+
+            if (optionKey.Option.Type == typeof(CodeStyleOption<bool>))
```

Comment:
It could. Since this was the only case that this appeared in, I decided to punt that problem for now. I anticipate such a thing will exist for .editorconfig stuff, since different options are going to want different text.